core/dbt/parser/read_files.py

```diff
@@ -17,11 +18,36 @@ def load_source_file(
     source_file = sf_cls(path=path, checksum=checksum,
                          parse_file_type=parse_file_type, project_name=project_name)
     source_file.contents = file_contents.strip()
-    if parse_file_type == ParseFileType.Schema:
-        source_file.dfy = yaml_from_file(source_file)
+    if parse_file_type == ParseFileType.Schema and source_file.contents:
+        dfy = yaml_from_file(source_file)
+        validate_yaml(source_file.path.original_file_path, dfy)
+        source_file.dfy = dfy
     return source_file
 
 
+# Do some minimal validation of the yaml in a schema file.
```

Comment:
Are we doing this validation on _all_ yaml files, or just the ones we've identified to be new/changed during partial parsing?

Reply:
I thought about this for a while, and in theory we could get a way with just doing it on changed and new files, but then the validation would have to be done in the partial parsing code, which felt wrong, and also yaml files with these errors wouldn't get any benefit from the better errors. Also we'd have different error messages depending on whether or not somebody had changed partial parsing.